clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1034,6 +1035,18 @@ public void run() {
                     pollTimeout = Math.min(pollTimeout, curHardShutdownTimeMs - now);
                 }
 
+                // For calls which have not yet been sent, update the node to send to if
+                // metadata has been refreshed. This is to handle controller change and
+                // node disconnections.
+                if (metadataManager.updater().lastMetadataUpdateMs() > lastIterationTimeMs) {
```

Comment:
Would it be more intuitive to put this logic in AdminClientRunnable.makeMetadataCall.handleResponse()? This allows us to reorder calls in `callsToSend` only when the MetadataResponse has no error. And we don't have to check timestamp to know whether the metadata has been updated.

Reply:
@lindong28 Thanks for the review. Yes, makes sense. Updated.